lib/common/xml.c

```diff
@@ -850,6 +850,13 @@ pcmk__xe_set_content(xmlNode *node, const char *format, ...)
         const char *content = NULL;
         char *buf = NULL;
 
+        /* xmlNodeSetContent() frees node->children and replaces it with new
+         * text. If this function is called for a node that already has a non-
+         * text child, it's a bug.
+         */
+        CRM_ASSERT((node->children == NULL)
+                   || (node->children->type == XML_TEXT_NODE));
```

Comment:
I'd rather free any existing content

Reply:
Making sure we're on the same page: We already do if it has existing text content. This is just a safeguard to avoid replacing a subtree of elements with text. For example, take `xml` to be the following:
```
<parent>
  <child some_attr="some_val"/>
<parent>
```

If we call `xmlNodeSetContent(xml, "some content")`, then that becomes
```
<parent>some content</parent>
```

with `<child>` getting freed.

Right now, there's nowhere in Pacemaker that we actually need to replace a subtree of elements with text. I figured if we do that, it's by accident.

---

In any case it's no trouble to add a `pcmk__xml_free(node->children)` to the top of this function (to free private data) and drop the assertion.

Reply:
Nope, I read it wrong. Fine as is, or maybe CHECK/return